backend/engine/processor/vulns.py

```diff
@@ -222,7 +237,7 @@ def _process_vuln_instance(
     vuln_scan, _ = VulnerabilityScanPlugin.objects.get_or_create(vuln_instance=vuln_instance, scan=scan)
     if not vuln_scan.plugins.filter(pk=plugin.pk).exists():
         vuln_scan.plugins.add(plugin)
-    if not vuln_scan.components.filter(pk=component.pk).exists():
+    if component is not None and not vuln_scan.components.filter(pk=component.pk).exists():
```

Comment:
Could this be simplified to:

```python
if component and not vuln_scan.components.filter(pk=component.pk).exists():
```

This would also catch if its an empty component as well. What do you think?

Reply:
That seems like a good idea. Those things should never happen, but just in case.